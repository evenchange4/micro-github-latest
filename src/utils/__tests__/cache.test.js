const cache = require('../cache');

it('should return cache store function', () => {
  expect(typeof cache.store).toBe('function');
});

it('should return cache get function', () => {
  expect(typeof cache.get).toBe('function');
});

it('should return correct store', () => {
  cache.store('evenchange4', 'micro-github-latest', { status: 302 });
  expect(cache.getAll()).toMatchSnapshot();

  cache.store('evenchange4', 'micro-medium-api', { status: 200 });
  expect(cache.getAll()).toMatchSnapshot();

  cache.store('evenchange4', 'micro-medium-api', { status: 201 });
  expect(cache.getAll()).toMatchSnapshot();
});
