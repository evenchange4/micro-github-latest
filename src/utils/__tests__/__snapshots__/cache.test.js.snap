// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return correct store 1`] = `
Object {
  "evenchange4/micro-github-latest": Object {
    "status": 302,
  },
}
`;

exports[`should return correct store 2`] = `
Object {
  "evenchange4/micro-github-latest": Object {
    "status": 302,
  },
  "evenchange4/micro-medium-api": Object {
    "status": 200,
  },
}
`;

exports[`should return correct store 3`] = `
Object {
  "evenchange4/micro-github-latest": Object {
    "status": 302,
  },
  "evenchange4/micro-medium-api": Object {
    "status": 201,
  },
}
`;
